src/hooks/useStakingH160.ts

```diff
@@ -0,0 +1,204 @@
+import { useStore } from 'src/store';
+import { computed, Ref, watch, ref } from 'vue';
+import { contractInstance, Staking } from 'src/config/web3';
+import dappsStakingContractAbi from 'src/config/web3/abi/dapps-staking-abi.json';
+import { $web3 } from 'boot/api';
+import { getDefaultEthProvider } from 'src/config/web3/utils';
+import BN from 'bn.js';
+
+export function useStakingH160(addressRef: Ref<string>) {
+  const store = useStore();
+  const stakingRef = ref();
+
+  // dApps Staking precompiled contract address
+  const PRECOMPILED_ADDR = '0x0000000000000000000000000000000000005001';
+
+  watch(
+    () => [$web3.value, addressRef.value],
+    async () => {
+      if ($web3.value && addressRef.value) {
+        const ci = contractInstance(
```

Comment:
Currently, the dApps Staking precompiled is only deployed on Shibuya I think. But right now I can access the dApps Page while logging in with my MetaMask on Shiden and Astar.
If there is no check for if the network supports the precompiled contract or not, please add that so it prevents users from staking.

FYI this is what I got when staking 500 ASTR tokens on Astar Network with MetaMask https://blockscout.com/astar/tx/0x4f0fbed708734ec75670c4ebaed2e3ba1ca8e692973faf7d0099009c5663cd53/internal-transactions

Reply:
resolved this by preventing other chains except Shibuya